src/hooks/useTokenMetadata.ts

```diff
@@ -1,11 +1,18 @@
-import { OptionalAddress, State } from '$/types'
+import { Address, OptionalAddress, State } from '$/types'
 import { useSelector } from 'react-redux'
 
-function selectTokenMetadata(tokenAddress: OptionalAddress) {
-    return ({ misc }: State) =>
-        tokenAddress ? misc.tokenMetadatas[tokenAddress.toLowerCase()] : undefined
+export function tokenMetadataCacheKey(tokenAddress: Address, tokenIds: string[]) {
```

Comment:
We now cache token metadatas by a combination of token address and all token ids. We could take it even further* but.. nah.

*) Cached `[0x…, tokenId1, tokenId2, tokenId3]` means we know metadatas of `[0x…]`, `[0x…, tokenId1]`, `[0x…, tokenId2]`, `[0x…, tokenId1, tokenId3]`, etc.